src/jdk.compiler/share/classes/com/sun/tools/javac/code/Lint.java

```diff
@@ -362,10 +363,28 @@ public enum LintCategory {
             map.put(option, this);
         }
 
+        /**
+         * Get the {@link LintCategory} having the given command line option, if any.
+         *
+         * @param option lint category option string
+         * @return corresponding {@link LintCategory}, or null if none
+         */
         public static LintCategory get(String option) {
             return map.get(option);
         }
 
+        /**
+         * Get the {@link LintCategory} having the given command line option.
+         *
+         * @param option lint category option string
+         * @return corresponding {@link LintCategory}
+         * @throws IllegalArgumentException if no such lint category exists
+         */
+        public static LintCategory forOption(String option) {
```

Comment:
Maybe we should call this "getOrThrow" ?

Reply:
Yeah, or perhaps the most modern & proper thing to do would be to have `LintCategory.get()` return `Optional<LintCategory>`.

This method is only used in a couple of places so that refactoring seems safe.

Updated in 3f792a8b650.